madara/crates/client/block_production/src/lib.rs

```diff
@@ -114,6 +163,8 @@ pub struct BlockProductionTask<Mempool: MempoolProvider> {
     l1_data_provider: Arc<dyn L1DataProvider>,
     current_pending_tick: usize,
     metrics: Arc<BlockProductionMetrics>,
+    production_mode: BlockProductionMode,
+    external_trigger: Option<Arc<tokio::sync::Notify>>,
```

Comment:
i would prefer a
struct TriggerCloseBlock; // unit struct
`mpsc::Receiver<TriggerCloseBlock>`

i think it makes more sense Notify is too low level i don't like it here, wdyt